src/storage/mvcc/txn.rs

```diff
@@ -59,7 +59,7 @@ impl<'a> MvccTxn<'a> {
 
     pub fn prewrite(&mut self, mutation: Mutation, primary: &[u8]) -> Result<()> {
         let key = mutation.key();
-        if let Some((_, commit)) = try!(self.reader.seek_write(&key, u64::max_value())) {
+        if let Some((commit, _)) = try!(self.reader.seek_write(&key, u64::max_value())) {
```

Comment:
No. Here we can simply reject the _Prewrite_ no matter old write is _Put_, _Delete_, or _Rollback_.